geps/gep-2649/index.md

```diff
@@ -0,0 +1,744 @@
+
+The following parts of GEP-713 also apply here. Inherited Policy Attachments:
+
+- MUST be their own CRDs (e.g. `TimeoutPolicy`, `RetryPolicy` etc),
+- MUST include both `spec` and `status` stanzas
+- MUST have the `status` stanza include a `conditions` section using the standard
+  upstream Condition type
+- MUST use the `targetRef` struct to specify their target resource.
+- MUST follow the naming requirements (MUST be named to clearly indicate that the
+kind is a Policy, and SHOULD use the `Policy` suffix at the end of the Kind and
+`policies` at the end of the Resource names).
+
+Additionally, Inherited Policy Attachment:
+
+- MAY specify a `defaults` stanza, an `overrides` stanza, or both. If it does not,
+  it MUST specify if its fields are defaults or overrides in each field's godoc.
+
+## Inherited Policy Attachment
+
+Because a Inherited Policy is a metaresource, it targets some other resource
+and _augments_ its behavior.
+
+But why have this distinct from other types of metaresource? Because Inherited
+Policy resources are designed to have a way for settings to flow down a hierarchy.
+
+_Any_ object that affects settings outside of the object that it attaches to is
+an Inherited Policy.
+
+Note that the same object may be have some properties of both an Inherited Policy
+_and_ a Direct Policy if it can attach to multiple points of a hierarchy, such
+as if the same Policy can be atttached to a Gateway (where it affects all Routes
+attached to that Gateway) or to a Route (where it affects only that Route).
+
+If a Policy _can be_ used as an Inherited Policy, it MUST be treated as an
+Inherited Policy, regardless of whether a specific instance of the Policy is
+only affecting a single object.
+
+This is because all of the rules about inheritance apply to the object, regardless
+of where it is attached in the hierarchy.
+
+Regardless of exactly how they are designed, settings in an Inherited Policy
+generally fall into two categories, Defaults and Overrides.
+
+Defaults set the default value for something, and can be overridden by the
+“lower” objects (like a connection timeout default policy on a Gateway being
+overridable inside a HTTPRoute), and Overrides cannot be overridden by “lower”
+objects (like setting a maximum client timeout to some non-infinite value at the
+Gateway level to stop HTTPRoute owners from leaking connections over time).
+
+Note that while both types of setting are valid, both also come with difficulties
+in communicating the currently active setting to the owners of the modified
+objects. These are discussed in more detail below.
+
+## Inherited Policy design guidelines
+
+Here are some guidelines for when to consider using a Inherited Policy object:
+
+* The settings or configuration are bound to one containing object, but affect
+  other objects attached to that one (for example, affecting HTTPRoutes attached
+  to a single Gateway, or all HTTPRoutes in a GatewayClass).
```

Comment:
Related to other longer comment thread, I think this may end up capturing more than intended. Often the GW API resources don't match closely with the underlying infra, so you maybe combining information from multiple layers in the underlying implementation. For example, GKE's implementation of the API takes information from Gateways and Routes when configuring [URLMaps](https://cloud.google.com/compute/docs/reference/rest/v1/urlMaps), so it would be ~impossible for _any_ potential policy we created to not meet this specific condition.

Reply:
This is only intended to capture resources _in the same hierarchy_, or that users _need to know about_ to understand the state of the system.

If, in GKE's Policy objects, the users _need to know about_ the URLMaps when looking at the Gateways and Routes that they own, then yes, they can't be Direct Attached Policy objects, even if they meet the other criteria.

Reply:
Added some extra words at the top to try and make this clearer.